pkg/aks/create.go

```diff
@@ -39,219 +38,105 @@ func CreateCluster(ctx context.Context, cred *Credentials, clusterClient service
-
-			if !hasAuthorizedIPRange(ipRange, aksCluster.APIServerAccessProfile.AuthorizedIPRanges) {
-				*aksCluster.APIServerAccessProfile.AuthorizedIPRanges = append(*aksCluster.APIServerAccessProfile.AuthorizedIPRanges, ipRange)
-			}
-		}
-	}
-
-	// Linux profile
-	if managedCluster.LinuxProfile != nil {
-		aksCluster.LinuxProfile = managedCluster.LinuxProfile
-	}
-
-	// Network profile
-	if managedCluster.NetworkProfile != nil {
-		np := managedCluster.NetworkProfile
-		if np.NetworkPlugin != "" {
-			aksCluster.NetworkProfile.NetworkPlugin = np.NetworkPlugin
-		}
-		if np.NetworkPolicy != "" {
-			aksCluster.NetworkProfile.NetworkPolicy = np.NetworkPolicy
-		}
-		if np.NetworkMode != "" {
-			aksCluster.NetworkProfile.NetworkMode = np.NetworkMode
-		}
-		if np.DNSServiceIP != nil {
-			aksCluster.NetworkProfile.DNSServiceIP = np.DNSServiceIP
-		}
-		if np.DockerBridgeCidr != nil {
-			aksCluster.NetworkProfile.DockerBridgeCidr = np.DockerBridgeCidr
-		}
-		if np.PodCidr != nil {
-			aksCluster.NetworkProfile.PodCidr = np.PodCidr
-		}
-		if np.ServiceCidr != nil {
-			aksCluster.NetworkProfile.ServiceCidr = np.ServiceCidr
-		}
-		if np.OutboundType != "" {
-			aksCluster.NetworkProfile.OutboundType = np.OutboundType
-		}
-		if np.LoadBalancerSku != "" {
-			aksCluster.NetworkProfile.LoadBalancerSku = np.LoadBalancerSku
-		}
-		// LoadBalancerProfile is not configurable in Rancher so there won't be subfield conflicts. Just pull it from
-		// the state in AKS.
-	}
-
-	// Service principal client id and secret
-	if managedCluster.ServicePrincipalProfile != nil { // operator phase is 'updating' or 'active'
-		aksCluster.ServicePrincipalProfile = managedCluster.ServicePrincipalProfile
-	}
-
-	// Tags
-	if managedCluster.Tags != nil {
-		aksCluster.Tags = managedCluster.Tags
+// newManagedCluster creates a new managed Kubernetes cluster.
+func newManagedCluster(ctx context.Context, cred *Credentials, workplacesClient services.WorkplacesClientInterface, spec *aksv1.AKSClusterConfigSpec, phase string) (*containerservice.ManagedCluster, error) {
+	managedCluster := &containerservice.ManagedCluster{
+		Name:     to.StringPtr(spec.ClusterName),
+		Location: to.StringPtr(spec.ResourceLocation),
+		ManagedClusterProperties: &containerservice.ManagedClusterProperties{
+			KubernetesVersion: spec.KubernetesVersion,
+		},
 	}
 
-	_, err = clusterClient.CreateOrUpdate(
-		ctx,
-		spec.ResourceGroup,
-		spec.ClusterName,
-		aksCluster,
-	)
-
-	return err
-}
-
-// createManagedCluster creates a new managed Kubernetes cluster.
-func createManagedCluster(ctx context.Context, cred *Credentials, workplacesClient services.WorkplacesClientInterface, spec *aksv1.AKSClusterConfigSpec, phase string) (containerservice.ManagedCluster, error) {
-	managedCluster := containerservice.ManagedCluster{}
-
-	// Get tags from config spec
-	tags := make(map[string]*string)
+	managedCluster.Tags = make(map[string]*string)
```

Comment:
This will cause a panic if `managedCluster.Tags` is nil and a key tries to get set.

In addition, if you remove

`managedCluster.Tags = make(map[string]*string)`

the controller will see that `managedCluster.Tags is nil` and will try to overwrite the Azure field with that state. I think you want to leave this as an empty map object even if no tags are added so that upstream tags are not removed on creation.